modules/internal/ChapelRange.chpl

```diff
@@ -2092,13 +2092,15 @@ proc _cast(type t, r: range(?)) where isRangeType(t) {
   //
   // source: Knuth Volume 2 --- Section 4.5.2
   //
-  proc chpl__extendedEuclidHelper(u, v)
+  proc chpl__extendedEuclid(u, v)
+    where isIntType(u.type) && isIntType(v.type)
   {
-    var zero: u.type = 0;
-    var one: u.type = 1;
+    // figure out the bigger type (in case u is int(64) v is int(32) e.g.)
+    var zero = 0:u.type + 0:v.type;
```

Comment:
I'm also curious what motivated these changes and whether there's a test that will exhibit the problem if it were reverted...